eclipse-collections-api/src/main/java/org/eclipse/collections/api/bag/MutableBag.java

```diff
@@ -84,16 +84,32 @@
     }
 
     @Override
-    MutableBag<T> with(T element);
+    default MutableBag<T> with(T element)
+    {
+        MutableBagIterable.super.with(element);
+        return this;
+    }
 
     @Override
-    MutableBag<T> without(T element);
+    default MutableBag<T> without(T element)
+    {
+        MutableBagIterable.super.without(element);
+        return this;
+    }
 
     @Override
-    MutableBag<T> withAll(Iterable<? extends T> elements);
+    default MutableBag<T> withAll(Iterable<? extends T> elements)
+    {
+        MutableBagIterable.super.withAll(elements);
+        return this;
+    }
 
     @Override
-    MutableBag<T> withoutAll(Iterable<? extends T> elements);
+    default MutableBag<T> withoutAll(Iterable<? extends T> elements)
+    {
+        MutableBagIterable.super.withoutAll(elements);
+        return this;
+    }
```

Comment:
Why do these call `super`?

Reply:
I tried to follow the same patter suggested by Donald here 
https://github.com/eclipse/eclipse-collections/issues/822#issuecomment-599230918

Reply:
> With these you can call super and return `this`

This was your suggestion in the linked comment @aboullaite pointed to above @motlin

Reply:
@motlin @nikhilnanivadekar please share your opinions on this question.  I presented two options, and a third was proposed which I agreed with and posted what the resulting solution would look like here:

https://github.com/eclipse/eclipse-collections/issues/822#issuecomment-599230918

Reply:
I find that every method which returns `this` delegates to another method and then returns `this`. They're pretty much always two lines. We can basically specify the implementation in Javadoc and have a consistent implementation everywhere. That's why I slightly prefer the implementation without `super`.

I realize this doesn't matter much and both implementations are fine. It's a slight preference. I'll go with whatever you'd like here.

Reply:
This is what the implementation will look like in MutableCollection:
```
default MutableCollection<T> with(T element)
{
    this.add(element);
    return this;
}
```
Duplicating the implementation looks like this:
```
@Override
default MutableList<T> with(T element)
{
    this.add(element);
    return this;
}
```
Or calling the super default implementation:
```
@Override
default MutableList<T> with(T element)
{
    MutableCollection.super.with(element);
    return this;
}
```
The question I have is what is the benefit of pulling up the implementation to `MutableCollection` if we are not going to use the pulled up implementation? This could create an uncovered or difficult to cover code path. We could leave it abstract in `MutableCollection` and enforce it being overridden at leaf interfaces. Thoughts?

Reply:
Good point. When I do work like this, I really do pull the methods up, using the IntelliJ refactoring. These methods aren't implemented in AbstractMutableCollection. If they were, I'd move them with the confidence that they are already covered. +1 these should be abstract on MutableCollection.

Reply:
So If I'm not mistaken, this would look something like this in MutableCollection
```
    abstract MutableCollection<T> withAll(Iterable<? extends T> elements);
```
and in other classes 
```
    @Override
    default MutableBagIterable<T> with(T element)
    {
        this.with(element);
        return this;
    }
```

Reply:
Correct, minus the abstract as `MutableCollection` is an interface.